src/ogc/wcs/WcsCoverage.js

```diff
@@ -72,13 +74,62 @@ define([
              * A configuration object for use by TiledElevationCoverage.
              * @type {{}}
              */
-            this.elevationConfiguration = this.createElevationConfiguration();
+            this.elevationConfig = this.createElevationConfig();
+        };
+
+        /**
+         * Preferred formats used for fuzzy comparison to available formats.
+         * @type {string[]}
+         */
+        WcsCoverage.PREFERRED_FORMATS = ["geotiff", "tiff"];
+
+        /**
+         * The default data format.
+         * @type {string}
+         */
+        WcsCoverage.DEFAULT_FORMAT = "image/tiff";
+
+        // Internal use only
+        WcsCoverage.prototype.createElevationConfig = function () {
+            return {
+                resolution: this.resolution,
+                coverageSector: this.sector,
+                retrievalImageFormat: this.determineFormatFromService(),
+                urlBuilder: new WcsUrlBuilder(this)
+            };
         };
 
         // Internal use only
-        WcsCoverage.prototype.createElevationConfiguration = function () {
-            // TODO
-            return {};
+        WcsCoverage.prototype.determineFormatFromService = function () {
+            var version = this.service.capabilities.version, availableFormats, format, coverageDescription;
+
+            // find the service supported format identifiers
+            if (version === "1.0.0") {
+                // find the associated coverage description
+                for (var i = 0, len = this.service.coverageDescriptions.coverages.length; i < len; i++) {
+                    if (this.coverageId === this.service.coverageDescriptions.coverages[i].name) {
+                        availableFormats = this.service.coverageDescriptions.coverages[i].supportedFormats.formats;
+                        break;
+                    }
+                }
+            } else if (version === "2.0.1" || version === "2.0.0") {
+                availableFormats = this.service.capabilities.serviceMetadata.formatsSupported
+            }
+
+            if (!availableFormats) {
+                return WcsCoverage.DEFAULT_FORMAT;
+            }
+
+            for (i = 0; i < WcsCoverage.PREFERRED_FORMATS.length; i++) {
```

Comment:
Order n-squared loop. How about storing preferred formats as a map?

Reply:
@pdavidc will you provide a short demo of what you mean?

Reply:
By map, I mean a data structure you can access by key, rather than by index. So the check to see if an available format matches a preferred format would be a constant time lookup.

Reply:
I could use a map, but I'm not sure how to incorporate the fuzzy comparison.